src/shared-components/accordion/test.tsx

```diff
@@ -13,40 +12,44 @@ const testAccordionProps = {
   children: <div>expansion</div>,
 };
 
-const AccordionWithTheme = (additionalProps?: Record<string, unknown>) => (
-  <ThemeProvider theme={primaryTheme}>
-    <Accordion {...testAccordionProps} {...additionalProps} />
-  </ThemeProvider>
-);
-
 describe('<Accordion />', () => {
+  const DecoratedAccordion = decorateWithThemeProvider(Accordion);
+
   test('renders regular accordion', () => {
-    const component = renderer.create(<AccordionWithTheme />);
+    const component = renderer.create(
+      <DecoratedAccordion {...testAccordionProps} />,
+    );
 
     const tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
 
   test('renders no border accordion', () => {
-    const component = renderer.create(<AccordionWithTheme noBorder />);
+    const component = renderer.create(
+      <DecoratedAccordion {...testAccordionProps} noBorder />,
+    );
 
     const tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
 
   test('renders disabled accordion', () => {
-    const component = renderer.create(<AccordionWithTheme disabled />);
+    const component = renderer.create(
+      <DecoratedAccordion {...testAccordionProps} disabled />,
+    );
 
     const tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
 
   test('invokes onClick when title is clicked', () => {
     const spy = jest.fn();
-    const titleIndex = 0;
 
-    const component = shallow(<AccordionWithTheme onClick={spy} />);
-    const title = component.childAt(titleIndex);
+    const component = mount(
+      <DecoratedAccordion {...testAccordionProps} onClick={spy} />,
+    );
+
+    const title = component.find('div[role="button"]');
```

Comment:
The way we were accessing the title was quite brittle, and adding a higher order component blew up the test. I relied on `mount` in conjunction with `find` to make it more robust.